app/modules/utils.py

```
@@ -1,16 +1,43 @@
     sleep_sec = (target - now).total_seconds()
+
     s = target.strftime("%d-%m-%Y %H-%M %Z")
     logging.info(f"sleeping until {s}")
+
     time.sleep(sleep_sec)
 
+
+def ensure_writable_dir(path: str):
+    os.makedirs(path, exist_ok=True)
+    if not os.access(path, os.W_OK):
+        raise PermissionError(f"Directory {path} is now writable")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)